### ISSUE TITLE
Test publish to real confluence

### ISSUE THREAD

**Issue by dbast** [docToolchain/docToolchain#376]
An idea that may help with confluence deploy.

**Is your feature request related to a problem? Please describe.**
There are some past and current issues related to confluence deploy.

**Describe the solution you'd like**
Test publish via CI to confluence could continuously test the confluence deploy mechanics:
* in Docker (maybe complicated but exits https://hub.docker.com/r/atlassian/confluence-server)
* real public space via TOKEN to a hidden page just to test the mechanics (hidden to avoid distributing un-reviewed content from PRs)... either a free dedicated space or maybe via https://arc42-template.atlassian.net/? ... that would require a token to be stored as env variable on travis.

**Additional context**
Make `publishToConfluence` almost a first class citizen as are `generatePDF` and `generateHTML` :)

**Comment by sabatmonk** [comment]
why i usually do is use a separate test space for branches and the real space in master.
i set a environment variable in the CI/CD pipeline and call it with System.getenv("SPACE") in the Config.groovy

**Comment by dbast** [comment]
Yeah, that is what I also do internally ... having that on the public here would be nice.

**Comment by rdmueller** [comment]
these are good ideas. I wasn't aware of the docker container. That might be a good solution - maybe paired with java testcontainers.
I currently try to rewrite the underlying architecture of docToolchain to make many tings better - this should be part of the general effort.
